ada/utils.py

```diff
@@ -1,8 +1,268 @@
+        explanations = ""
+    if "options" in body.keys() and cardType != "CHOICE":
+        optionId = body["options"][0]["optionId"]
+    else:
+        optionId = None
+    path = body["_links"]["next"]["href"]
+    if "step" in body.keys():
+        step = body["step"]
+    else:
+        step = ""
+    if cardType == "CHOICE":
+        option = body["options"][0]["text"]["en-GB"]
+        optionslist = []
+        index = 0
+        length = len(body["options"])
+        while index < length:
+            optionslist.append(body["options"][index]["text"]["en-GB"])
+            index += 1
+        choices = "\n".join(optionslist)
+        extra_message = (
+            f"Choose the option that matches your answer. Eg, 1 for {option}"
+        )
+        message = f"{description}\n\n{choices}\n\n{extra_message}\n\n{back}"
+        body = {}
+        body["choices"] = length
+    elif cardType == "TEXT":
+        message = f"{description}\n\n{textcontinue}\n\n{back}"
+        body = {}
+    else:
+        message = f"{description}\n\n{back}"
+        body = {}
+        body["choices"] = None
+    body["message"] = message
+    body["explanations"] = explanations
+    body["step"] = step
+    body["optionId"] = optionId
+    body["path"] = path
+    body["cardType"] = cardType
+    body["title"] = title
+    body["description"] = description
+    return body
+
+
+def get_endpoint(payload):
+    head = {"Content-Type": "application/json"}
+    value = payload["value"]
+    if value != "":
+        if value == "back":
+            url = reverse("ada-previous-dialog")
+            response = requests.post(url, data=json.dumps(payload))
+            return response
+        elif value == "abort":
+            url = reverse("ada-abort")
+            response = requests.post(url, data=json.dumps(payload))
+            return response
+        else:
+            url = reverse("ada-next-dialog")
+            response = requests.post(url, data=json.dumps(payload), headers=head)
+            return response
+    elif value == "":
+        url = reverse("ada-start-assessment")
+        data = json.dumps(payload)
+        response = requests.post(url, data, headers=head)
+        return response
+
+
+def get_path(body):
+    if "path" in body.keys():
+        path = body["path"]
+    else:
+        path = body["_links"]["startAssessment"]["href"]
+    return path
+
+
+def get_step(body):
+    step = body["step"]
+    return step
+
+
+def pdf_ready(data):
+    try:
+        data["_links"]["report"]["href"]
+        return True
+    except KeyError:
+        return False
+
+
+def pdf_endpoint(data):
+    head = {"Content-Type": "application/json"}
+    url = reverse("ada-reports")
+    response = requests.post(url, data=json.dumps(data), headers=head)
+    return response
+
+
+# This returns the report of the assessment
+def get_report(data):
+    head = {
+        "x-ada-clientId": settings.X_ADA_CLIENTID,
+        "x-ada-userId": settings.X_ADA_USERID,
```

Comment:
I thought one of these two need to be the msisdn or something specific to the user?

I also see we repeat this a lot, maybe we can define it in one place and grab it from there?

Reply:
That should be the assessmentid but it is returned as a response in ADA's payload. Something like this:

"next": {
                    "method": "POST",
                    "href": (
                        "/assessments/"
                        "654f856d-c602-4347-8713-8f8196d66be3/dialog/next"
                    ),
                },
                
                
 However, I realized the ID wasn't being saved so the latest pr saves it.

Reply:
That should be the assessmentid but it is returned as a response in ADA's payload. Something like this:

"next": {
                    "method": "POST",
                    "href": (
                        "/assessments/"
                        "654f856d-c602-4347-8713-8f8196d66be3/dialog/next"
                    ),
                },
                
                
 However, I realized the ID wasn't being saved so the latest pr saves it.